### DISCUSSION TITLE
How to stream mono using ffmpeg

### DISCUSSION THREAD

**Discussion by nasirq** [savonet/liquidsoap#2667] [User Support]
We are using Liquidsoap 2.1.2
And are unable to setup to stream mono. Any help is appreciated.

**Comment by rmens** [comment]
You might need to convert your stereo input to mono audio first. You can use the `mean` operator for that. For example:

`stereoinput = single('/var/audio/hello.ogg')`

make it mono:

`monoaudio = mean(stereoinput)`

And feed ffmpeg your `monoaudio` source.

**Reply by toots** [reply]
@rmens solution could work but we are also able to handle mono conversion at the encoder level. This should work:

```ruby
s = ...

output.icecast(
  ...,
  %ffmpeg(format="mp3", %audio(codec="libmp3lame", channels=1),
  s
)
```
